src/main/groovy/org/gradle/android/workarounds/CompilerArgsProcessor.groovy

```diff
@@ -40,12 +40,10 @@ class CompilerArgsProcessor {
         }
         applied = true
 
-        project.tasks.withType(JavaCompile).configureEach { JavaCompile task ->
-            project.gradle.taskGraph.beforeTask {
-                if (task == it) {
-                    def processedArgs = processArgs(task.options.compilerArgs, task)
-                    overrideProperty(task, processedArgs)
-                }
+        project.gradle.taskGraph.whenReady {
```

Comment:
i think @ghale would be the best person to review if this still accomplishes the same goal

Reply:
Actually, I think we don't need `CompilerArgsProcessor` anymore.  I think we got rid of the last classes that use it when we removed support for some of the older AGP versions.  We should be able to just remove this class, its test, and even `WorkaroundContext`.

Reply:
I've removed all of the above, thanks.